wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt

```diff
@@ -636,7 +637,11 @@ class MainViewModel @Inject constructor(
                 // update the current item by replacing the current item
                 items[dateKey]?.toMutableList()?.let { list ->
                     val itemIndex = list.indexOfFirst { it.id == rowView.id }
-                    list[itemIndex] = transactionRow
+                    if (itemIndex == -1) {
+                        log.info("cannot find {} in list of {} items", rowView.id, list.size)
+                    } else {
+                        list[itemIndex] = transactionRow
+                    }
```

Comment:
not sure what causes this, but it does crash the app.  We can log it and see if we can find the cause.